lib/cog_api/http/base.ex

```diff
@@ -128,11 +127,19 @@ defmodule CogApi.HTTP.Base do
       "#{url}/#{route}"
     end
 
-    if map_size(params) == 0 do
-      url
-    else
-      URI.encode(url <> "?" <> URI.encode_query(params))
-    end
+    format_params(url, params)
+  end
+
+  defp format_params(url, params) when is_map params and map_size(params) == 0 do
+    url
+  end
+
+  defp format_params(url, params) when is_map params do
+    URI.encode(url <> "?" <> URI.encode_query(params))
+  end
+
+  defp format_params(url, params) do
```

Comment:
https://github.com/operable/cog-api-client/blob/jd-ae-roles-show/lib/cog_api/fake/client.ex#L16


Any things that is not a map (and id in this case).